log/ratelimit.go

```diff
@@ -0,0 +1,116 @@
+package log
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"golang.org/x/time/rate"
+)
+
+const (
+	infoLevel  = "info"
+	debugLevel = "debug"
+	warnLevel  = "warning"
+	errorLevel = "error"
+)
+
+type RateLimitedLogger struct {
+	next    Interface
+	limiter *rate.Limiter
+
+	discardedLogLinesCounter *prometheus.CounterVec
+}
+
+// NewRateLimitedLogger returns a logger.Interface that is limited to the given number of logs per second,
+// with the given burst size.
+func NewRateLimitedLogger(logger Interface, logsPerSecond rate.Limit, burstSize int, reg prometheus.Registerer) Interface {
+	discardedLogLinesCounter := promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+		Name: "rate_limit_logger_discarded_log_lines_total",
```

Comment:
[nit] I would just use `logger_` as prefix.

```suggestion
		Name: "logger_rate_limit_discarded_log_lines_total",
```